tests/matrixstore/test_web.py

```diff
@@ -123,3 +123,103 @@ def test_import() -> None:
+    refs_one = [ref_one, ref_two]
+    refs_two = [ref_three]
+    refs_three = [ref_four]
+
+    disk_usage = 24
+    matrix_reference_one = MatrixReferencesDTO(refs=refs_one, disk_usage=disk_usage)
+    matrix_reference_two = MatrixReferencesDTO(refs=refs_two, disk_usage=disk_usage)
+    matrix_reference_three = MatrixReferencesDTO(refs=refs_three, disk_usage=disk_usage)
+    matrix_dict = {
+        command_id: matrix_reference_one,
+        "constant_id_1": matrix_reference_two,
+        "test": matrix_reference_three,
+    }
+    service = Mock()
+    matrix_references = {
+        MatrixReference(matrix_id=command_id, use_description=f"Referenced by command {command_id} of study study1"),
+        MatrixReference(matrix_id=command_id, use_description=f"Referenced by raw study {study_id}"),
+        MatrixReference(matrix_id="constant_id_1", use_description="Constant matrix"),
+        MatrixReference(matrix_id="test", use_description=f"Referenced by dataset {dataset_id}"),
+    }
+    service.get_used_matrices.return_value = matrix_references
+
+    service.get_matrices_references.return_value = matrix_dict
+    app = create_app(service)
+    client = TestClient(app)
+    res = client.get("/v1/matrix/_references/", headers=create_auth_token(app), params={"disk_usage": True})
+
+    res_dict_test = creating_json_res_dict(res)
+
+    sorted_matrix_dict = dict(sorted(matrix_dict.items()))
+    sorted_res_matrix_dict = dict(sorted(res_dict_test.items()))
+    assert res.status_code == 200
+    assert sorted_res_matrix_dict == sorted_matrix_dict
+
+    matrix_reference_one = MatrixReferencesDTO(refs=refs_one, disk_usage=None)
+    matrix_reference_two = MatrixReferencesDTO(refs=refs_two, disk_usage=None)
+    matrix_reference_three = MatrixReferencesDTO(refs=refs_three, disk_usage=None)
+    matrix_dict = {
+        command_id: matrix_reference_one,
+        "constant_id_1": matrix_reference_two,
+        "test": matrix_reference_three,
+    }
+
+    service.get_used_matrices.return_value = matrix_references
+
+    service.get_matrices_references.return_value = matrix_dict
+
+    res = client.get("/v1/matrix/_references/", headers=create_auth_token(app), params={"disk_usage": False})
+    assert res.status_code == 200
+    data__ = {
+        f"{command_id}": {
+            "refs": [
+                {"description": f"Referenced by command {command_id} of study study1"},
+                {"description": "Referenced by raw study study_id"},
+            ]
+        },
+        "constant_id_1": {"refs": [{"description": "Constant matrix"}]},
+        "test": {"refs": [{"description": "Referenced by dataset data_1"}]},
+    }
+    assert res.json() == data__
+
+    assert "disk_usage" not in res.json()[command_id]
+    assert "disk_usage" not in res.json()["constant_id_1"]
+    assert "disk_usage" not in res.json()["test"]
+
+    res_dict_test = creating_json_res_dict(res)
```

Comment:
Same, I don't think we need all the following code since we have already checked the full result